### DISCUSSION TITLE
Github - Projects (classic) sunset

### DISCUSSION THREAD

**Discussion by Torinde** [dosbox-staging/dosbox-staging#3900] [General]
[Projects (classic) is officially sunset](https://github.blog/changelog/2024-05-23-sunset-notice-projects-classic/)
> August 23, 2024: Projects (classic) will be officially sunset. All unmigrated classic projects will automatically be migrated to new projects.
If projects is disabled on the organization level, classic projects under that organization will not be migrated.
Only cards that have been updated in the last year will be migrated.
During the migration process, projects v2 webhooks for the newly created projects will not be emitted.

There is a "start migration" button, but **I haven't clicked it yet.**

Cards that haven't been updated last year will not be migrated, which is a pity, since some may go missing then...

Backlog Planned features
- Write documentation about performance tuning: how to create flame graphs and icicle graphs. Added by [dreamer](https://github.com/dreamer)

Backlog Serious problems to be resolved
- #342

Backlog Suggested new features
- #313 
- #686 
- #316 
- #946 
- #1045
- #947 
- #1789 
- #1265 
- #2138 
- #2224 
- #1792 
- #1118 
- #2826 
- #2973 
- #583 
- #3115

Backlog Low priority bugs
- #610 

Backlog Done, with status "completed", but they aren't completed (I think that actually means "longer term goal, not of immediate concern")
- #1095
- #565
- for sure there are others

edit: Other notable items (not from Projects):
- #1953 
- #2784 
- https://github.com/dosbox-staging/dosbox-staging/discussions/2054#discussioncomment-8581995
- #1864
- #1262
- #3156

eXoDOS features
- #1682 
- #2619
- #1671 
  - [8 games supporting GunStick](https://computeremuzone.com/pc/subgen/gunstick&l=en) - there is a small DOSbox patch that allows to use the mouse for that
  - https://github.com/dosbox-staging/dosbox-staging/issues/4426, alternative solution: #3563
- #3635 
- #2903 
- #2904 
- #2906 
- #2908 
- #2900 
- #3737
- #3738 
- #3739 

From closed projects, which may be still unfinished
- #1720 
- #1168 
- #716 
- #92
- #1274

Backlog Rejected features
- #332
- #392 
- #603 
- #402 
- #209 
- #99
- #399 
- #661 
- #166
- #256
- #612 
- #391 
- #1073 
- #1040 
- #1081 
- You can't see this card Added by [nemo93 ](https://github.com/nemo93)on Aug 26, 2021
- #1508

Backlog Done, but with status "not planned" (so, similar to "rejected"?)
- #100 
- #62
- #2131 
- #904 
- #664

eXoDOS done, cards without issue (not including such with minimal info/just label that's obviously completed such as "MT-32")
- Add PCjr composite support and improved colours: https://www.vogons.org/viewtopic.php?t=12319&start=680 https://www.vogons.org/viewtopic.php?p=677941#p677941 Added by [kcgen](https://github.com/kcgen)
- IDE interface - Some games will only run from inside a virtual hard disk. For those games to work, they are first setup by booting into a VHD formatted image and setup using real DOS. IDE is needed after the fact in dosbox to mount the CD image along-side the HDD image. European Racers (1993), Operation AirStorm (1994), Pilgrim Quest (1991), Backroad Racers (1993). Windows 3.1 games that have problems launched from files and need disk image: https://github.com/exoscoriae/eXoWin3x/labels/VHD%20Boot IDE support is needed more-so in win3x software, however any game using real DOS via VHD that needs to mount a CD image needs it as well. Added by [kcgen](https://github.com/kcgen)
- Sensible Golf (two issues): There is a speed issue which is fixed by switching to X (ECE gives error 21)
The mouse cursor is invisible. It you alt tab and then click back into the dosbox window you get the cursor back but as soon as you click on the game window inside dosbox it disappears again. Added by [kcgen](https://github.com/kcgen)
- Better DAC emulation - Example: Crystals of Arborea (1990). Intro logo has a big beep sound but in X its a proper tone. Inside the game if you select "3D", you get a terrible choked lizard sound, in X its bird sounds. ... Tandy Skate or Die doesn't sound as good as on real hardware. ... If you load "John madden football II" with the switch "MFB2 RA" for Roland with adlib you get sampled adlib sound which is poor. Another is Battle Chess if loaded with "/adlib" switch. Added by [kcgen](https://github.com/kcgen)

Done/Archived in closed projects, cards without issue
- Implement mixer /listmidi on ALSA and use the implementation to automatically connect to the appropriate port. Added by [dreamer](https://github.com/dreamer)

ECE patches Done
- Enable 4 MB video memory, hopefully improving Build Engine games. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/20cc060af4d34852b4800b95e2d5bc03fd063175 Make memory size configurable, 512k (800x600 at 256 colors), 1 MiB (1024x768 at 256, 800x600 at 64k), 2 MiB (1600x1200 at 256, 1024x768 at 64k, 640x480 at 16M colors), 4 MiB (1600x1200 at 64k, 1024x768 at 16M), 8 MiB: (1024x768 at 16M double-buffered, or 1600x1200 at 16M), for build engine games. Thanks to Xavier_Velo for this list.
- Allow mapping keys to mouse buttons. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/d88f6db3b9193d276756b01703c6e9822e8646e5
- Add Voodoo1 emulation and Glide API pass-through. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/594e3209e38e5bcf34f772df9164c7f1f793fafb We're dropping Glide pass-through and focusing only on 3dfx Voodoo emulation for now. https://github.com/dosbox-staging/dosbox-staging/issues/339
- Improve PC speaker emulation. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/82f36a265c2a22b4ce6aae352cb227893e8f9ed4 Update Mar-2021: ripa on Discord has started working on this to prepare it for PR.
- Enable a maximum of 256 MB of system memory, hopefully improving Windows 9x running on top of DOSBox ECE. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/c486efacdad7cea5e0888e4b26cc970cbc3f3acc It seems like this patch is incorrect or incomplete, but I'm not 100% sure yet - the only reference to implemented INT15 E801h I found is: https://www.techarp.com/bios-guide/os-select-dram-64mb/ - - according to that page it applies to older, unpatched versions of OS/2 only. DOSBox-X includes this patch and extends it with handling of INT15 E820h - that's what Windows is supposedly using.
- Add Fluidsynth support. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/f786b9fdc52df0e9b39ee27d640fb37ba7c369e6 We're going with a modernized version of this patch for FluidSynth 2.x integration: https://github.com/dosbox-staging/dosbox-staging/issues/262 Ported in: 0.76.0
- Add mt32emu MIDI device. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/bdcf933ae9989515d6381d1907077db273c220dd https://github.com/dosbox-staging/dosbox-staging/issues/257
- Add support for more than 4 axes and digipads. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/b881409bd89e36bc0b140ba84ad90e39258361fa https://github.com/dosbox-staging/dosbox-staging/pull/403 Ported in: 0.76.0, Backported to: 0.75.1
- Implement pixel-perfect scaling. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/d66ef499fde5db07b703334a37be2ede8a04cacb Released in 0.75.0
- Add Nuked OPL3 emulator. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/6cfaad2ce9ef7363b1093fd068dc8d497a1a49a9 Released in 0.75.0
- Add support for FLAC, Opus, and MP3 audio tracks. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/9aec7c324fa047da584d0c034d792e1e6064866f Released in 0.75.0

ECE patches Rejected
- Add scalers normal4-6x. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/e2c368de27b69e3d1d91dd60744ae44aa256b185 Rejected, as this change is pointless since there's support for sharp bilinear scaling using glshaders.
- Change the icon and splash screen. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/440e4c44d923da89e1b9bd3e983be20e8d7a2bb6 Rejected, as this is ECE specific; we have our own splash screen and icon.
- Add Qbix' patch for a more accurately balanced SoundBlaster volume. ECE patch: https://github.com/dosbox-staging/dosbox-staging/commit/421088a6628f00b38c2d90372ca216e0c89a2a1f

From closed projects, finished subsequently:
- #1577, alternative: DEVICE.COM in #3954
- #1214, done: https://github.com/dosbox-staging/dosbox-staging/pull/1267#issuecomment-922391196 (eXoDOS archived)

Accessories packs (not from projects):
- #3954 - Support files that almost everyone needs
- #3895 - Benchmarking and system functionality testing toolkit
- #4300
- [Ethernet pack](https://github.com/dosbox-staging/dosbox-staging/pull/4330) 

I didn't list all the "Done" items (for Backlog, eXoDOS and for the closed projects) - is it useful to keep such record?

**Comment by Grounded0** [comment]
I think almost all of the stuff have been ticketised from GitHub Projects now. Thanks for helping on this one.

**Reply by Torinde** [reply]
There are three pending actions from above:
- [x] I don't see actual issue for "Write documentation about performance tuning: how to create flame graphs and icicle graphs."
- [ ] Most of the "Done" items (for Backlog, eXoDOS and for the closed projects) will be lost when migrating, per my understanding. Is it useful to have those listed here before that?
- [ ] Push "start migration" button, when sure no more of the old items are relevant
![image](https://github.com/user-attachments/assets/f2e2de0c-8c1a-47f5-a713-15a80b5fb6cf)

**Reply by Grounded0** [reply]
I think DOSBENCH covers performance profiling needs for this project well so we can disregard that. Dreamer is retired from the project.

The only open progress report we have is now fully covered by label https://github.com/dosbox-staging/dosbox-staging/labels/eXoDOS%2FWin3x so we can disregard that too.

**Reply by Torinde** [reply]
> The only open progress report we have is now fully covered by label [eXoDOS/Win3x](https://github.com/dosbox-staging/dosbox-staging/labels/eXoDOS%2FWin3x) so we can disregard that too.

My second question is not about that. It's about whether it's relevant to keep the list of issues implemented ("Done") as part of Backlog/eXoDOS/0.78/0.77/etc. Per what I read it seems if you migrate the classic projects, then entries older than 1 year will be gone (Issues themselves will remain, but will not be "linked" anymore to the migrated project).

Some lists that may disappear after migration:
- [ ] Backlog - 124 Done items
- [x] eXoDOS - there's a label as you say, so no loss
- [x] ECE - I listed everything above, not sure if adding label is worth it
- [x] Ongoing tasks - 3 items which are also part of other projects, not sure if adding label is worth it
- [ ] 0.75.0 - 36 items
  - Splash screen needs to start in fullscreen, when sdl.fullscreen = true is selected. https://github.com/dosbox-staging/dosbox-staging/issues/178 and https://github.com/dosbox-staging/dosbox-staging/issues/180 are closely related from implementation perspective.
  - the cards without issues look like a release checklist - I assume that's covered elsewhere now - #2998, [Release-process](https://github.com/dosbox-staging/dosbox-staging/wiki/Release-process)?
- [ ] 0.75.1 - 17 items
- [ ] 0.75.2 - 23 items
- [ ] 0.76.0 - 75 items
- [ ] 0.76.1 - 6 items
- [ ] 0.77.0 - 84 items
- [ ] 0.77.1 - 5 items
- [ ] 0.78.0 - 142 items - some of these have tag for "Milestone 0.78"
- [ ] 0.78.1 - 13 items
- [ ] 0.79.0 - 290 items

**Reply by Grounded0** [reply]
I'll hand over to @johnnovak to decide can we accept this potential loss. I think we should be good as anything in progress would be currently saved.

**Reply by Torinde** [reply]
OK. For the items from release number projects - the number can be added as "Milestone" in the Issue/PR:
![image](https://github.com/user-attachments/assets/3460c342-ba54-482f-b3b9-36b3f8bf3361)
![image](https://github.com/user-attachments/assets/13f92902-4e08-43be-895b-e1cb473c0eb8)

Some of the 0.78 items have such tag:
![image](https://github.com/user-attachments/assets/0a0e0d1d-674e-47cf-9b26-52e3f3cb55d4)

My account doesn't have such rights, but maybe yours has.

**Reply by johnnovak** [reply]
@torinde @Grounded0 I'll deal with these at the end of Sept when I'm back from my holiday, thanks.

**Reply by Grounded0** [reply]
Seems like we lost all the data on GitHub Projects (Classic). We probably should start using milestones from 0.83 onwards to make sure this doesn't happen again.

**Reply by johnnovak** [reply]
> Seems like we lost all the data on GitHub Projects (Classic). We probably should start using milestones from 0.83 onwards to make sure this doesn't happen again.

Huh? All data is there, just it has been converted to the new format. Check out the projects tab, nothing is lost as far as I can tell.

Besides, even if the project views were lost (but they're not), the individual tickets would still remain. I don't think the project views have much value, I almost never look at them.

Also, we don't do milestones and planned releases here—everybody works on what they feel like, then we let the the scope of the next release unfold dynamically and organically 😎🤘🏻 Which is how it should be in a hobby project people do for fun in their spare time.

For the record, this is what happened when I clicked on our projects—they have been auto-converted:

<img width="601" alt="image" src="https://github.com/user-attachments/assets/14b5f1af-8233-4ad0-ab3d-f72c14ff977d">

**Comment by johnnovak** [comment]
@Torinde 

> There is a "start migration" button, but I haven't clicked it yet.

Well I clicked it, as per above, and it worked 😄 All information has been retained in the new project structure which is like 90% the old one anyway.

Lesson to learn: you could have spared yourself all this exercise by asking us first, before creating that megapost...

**Reply by Torinde** [reply]
Maybe myself (and @Grounded0) miss something, but what I see now only 3 projects (all closed projects are gone and there's no "Projects (classic)" selector where previously you could find them):
![image](https://github.com/user-attachments/assets/9d799965-001b-4e12-b998-5e3591f2f08b)

And they kept only the recent items (e.g. Backlog has 3 open items - before it had the list from my comment on the top here)
> Cards that haven't been updated last year will not be migrated, which is a pity, since some may go missing then...

That's how I understood the Github notification and it seems that's what happened?

I understand you don't find Projects tab so useful and prefer to manage the work via Issues/PRs/etc., but in case all the info is really lost (e.g. there isn't some admin/other view that I haven't seen), then I don't mind having done the above exercise :) - at least parts of the data is here now. Luckily, the [missing things](https://github.com/dosbox-staging/dosbox-staging/discussions/3900#discussioncomment-10453691) doesn't seem crucial?

**Reply by johnnovak** [reply]
@Torinde 

Indeed, I can't see them now in the project view. Interesting.

Well, like I said, the issue tickets are still there, intact. Just the project views are gone, which we don't care about.

Sure, you can spend your time however you like, but we told you previously a few times we don't care about the project views 🤷🏻 😄 

What I care about is tagging issues appropriately, with multiple somewhat overlapping tags, which gives you dynamic filtering abilities. To me that is 100x more useful than the "static" project views that you need to manually manage. Plus we're not operating like a company with deadlines, etc, so I don't find that classic project view useful at all. I've got enough of that at work, thank you 😅

**Reply by Torinde** [reply]
> overlapping tags, which gives you dynamic filtering abilities

Agree!

> Just the project views are gone

- also gone are the project "cards" which weren't an Issue/PR. Most of these I copied above (except the 'release checklist' ones), so that should be OK too.
- also gone is the _association_ of Issues/PRs per Staging release (and Backlog Done), which was too much for me to do - without rights to use the Milestone tags.

> I've got enough of that at work

Good luck! :)

**Comment by Torinde** [comment]
[0.81 Project](https://github.com/orgs/dosbox-staging/projects/2) has to be closed now?

**Reply by johnnovak** [reply]
@Torinde I've just closed it, but I find strange that you care after I explained we don't even look much at the projects...
